checks.d/marathon.py

```diff
@@ -59,30 +62,52 @@ def check(self, instance):
                         self.gauge('marathon.' + attr, app[attr], tags=tags)
 
         # Number of running/pending deployments
-        response = self.get_json(urljoin(url, "v2/deployments"), timeout, auth)
+        response = self.get_json(urljoin(url, "v2/deployments"), timeout, auth, acs_url)
         if response is not None:
             self.gauge('marathon.deployments', len(response), tags=instance_tags)
 
-    def get_json(self, url, timeout, auth):
+    def get_json(self, url, timeout, auth, acs_url):
+        params = {
+            'timeout': timeout,
+            'headers': {},
+            'auth': auth
+        }
+        if acs_url:
+            try:
+                auth_body = {
+                    'uid': auth[0],
+                    'password': auth[1]
+                }
+                r = requests.post(urljoin(acs_url, "acs/api/v1/auth/login"), json=auth_body, verify=False)
```

Comment:
Ok, added some logic to cache the acs token after it is grabbed for the first time, and only attempt to refresh when the token fails to auth.